dds/DCPS/security/AccessControl/Permissions.cpp

```diff
@@ -227,29 +232,94 @@ int Permissions::load(const SSL::SignedDocument& doc)
           ACE_TEXT("AccessControlBuiltInImpl::load_permissions_file: ")
           ACE_TEXT("Unable to parse subject name, ignoring grant.\n")));
       }
-    } else if (contains_subject_name(rule_holder_.subject)) {
+    } else if (find_grant(grant->subject)) {
       if (DCPS::security_debug.access_warn) {
         ACE_ERROR((LM_WARNING, ACE_TEXT("(%P|%t) {access_warn} ")
           ACE_TEXT("AccessControlBuiltInImpl::load_permissions_file: ")
           ACE_TEXT("Ignoring grant with duplicate subject name.\n")));
       }
     } else {
-      perm_data_.perm_rules.push_back(rule_holder_);
+      grants_.push_back(grant);
     }
   } // grant_rules
 
   return 0;
 }
 
-bool Permissions::contains_subject_name(const SSL::SubjectName& name) const
+bool Permissions::find_grant(const SSL::SubjectName& name, Grant_rch* found) const
```

Comment:
Should this just return the grant?

Reply:
caller doesn't need it in all cases, but it may be worth simplifying